src/modules/uuv_pos_control/uuv_pos_control.cpp

```diff
@@ -90,61 +90,143 @@ void UUVPOSControl::parameters_update(bool force)
-		const float roll_des, const float pitch_des, const float yaw_des,
-		vehicle_attitude_s &vehicle_attitude, vehicle_local_position_s &vlocal_pos)
+void UUVPOSControl::generate_trajectory_setpoint(vehicle_local_position_s &vlocal_pos,
+		vehicle_attitude_s &vehicle_attitude,
+		float dt)
 {
-	//get current rotation of vehicle
+	// Avoid accumulating absolute yaw error with arming stick gesture
+	float roll = Eulerf(matrix::Quatf(vehicle_attitude.q)).phi();
+	float pitch = Eulerf(matrix::Quatf(vehicle_attitude.q)).theta();
+	float yaw = Eulerf(matrix::Quatf(_trajectory_setpoint.quaternion)).psi();
+
+	// Integrate manual control inputs
+	// Info:
+	//   - only yaw is updated in position mode, roll and pitch are uncontrolled
+	//   - throttle is Z, roll is Y, pitch is X
+	float roll_setpoint = roll;
+	float pitch_setpoint = pitch;
+
+	if (_param_stab_mode.get()) {
+		roll_setpoint = 0.0;
+		pitch_setpoint = 0.0;
+	}
+
+	float yaw_setpoint = yaw + _manual_control_setpoint.yaw * dt * 0.5f;
+
+	// Update position setpoints based on manual control inputs
+	float vx_sp = 0.0;
+
+	if (_manual_control_setpoint.pitch > _param_pos_stick_db.get()
+	    || _manual_control_setpoint.pitch < -_param_pos_stick_db.get()) {
+		// If pitch is not zero, we use it to set the roll setpoint
+		vx_sp = _manual_control_setpoint.pitch * _param_pgm_vel.get();
+	}
+
+	float vy_sp = 0.0;
+
+	if (_manual_control_setpoint.roll > _param_pos_stick_db.get()
+	    || _manual_control_setpoint.roll < -_param_pos_stick_db.get()) {
+		// If roll is not zero, we use it to set the pitch setpoint
+		vy_sp = _manual_control_setpoint.roll * _param_pgm_vel.get();
+	}
+
+	float vz_sp = 0.0;
+
+	if (_manual_control_setpoint.throttle > _param_pos_stick_db.get()
+	    || _manual_control_setpoint.throttle < -_param_pos_stick_db.get()) {
+		// If throttle is not zero, we use it to set the vertical velocity
+		vz_sp = -_manual_control_setpoint.throttle * _param_pgm_vel.get();
+	}
+
+	// rotate velocity setpoint in body frame to global frame
+	Vector3f velocity_setpoint(vx_sp, vy_sp, vz_sp);
 	Quatf q_att(vehicle_attitude.q);
+	Vector3f rotated_velocity_setpoint = q_att.rotateVector(velocity_setpoint);
+
+	// Generate target quaternion
+	Eulerf euler_sp(roll_setpoint, pitch_setpoint, yaw_setpoint);
+	Quatf q_sp = euler_sp;
+
+	// Normalize the quaternion to avoid numerical issues
+	q_sp.normalize();
 
-	Vector3f p_control_output = Vector3f(0,
-					     0,
-					     _param_pose_gain_z.get() * (pos_des(2) - vlocal_pos.z));
-	//potential d controller missing
-	Vector3f rotated_input = q_att.rotateVectorInverse(p_control_output);//rotate the coord.sys (from global to body)
+	q_sp.copyTo(_trajectory_setpoint.quaternion);
 
-	publish_attitude_setpoint(rotated_input(0) + pos_des(0), rotated_input(1) + pos_des(1), rotated_input(2),
-				  roll_des, pitch_des, yaw_des);
+	_trajectory_setpoint.timestamp = hrt_absolute_time();
 
+	if (_param_pos_mode.get()) {
```

Comment:
I think this needs to be `if (!_param_pos_mode.get())` if 1 it should be body, and not world.